pkg/suggestion/v1alpha3/hyperopt_service.py

```diff
@@ -40,17 +41,81 @@ def GetSuggestions(self, request, context):
         )
 
     def ValidateAlgorithmSettings(self, request, context):
+        is_valid, message = OptimizerConfiguration.validate_algorithm_spec(
+            request.experiment.spec.algorithm)
+        if not is_valid:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details(message)
+            logger.error(message)
         return api_pb2.ValidateAlgorithmSettingsReply()
 
 
-class OptimizerConfiguration(object):
-    def __init__(self, random_state=None):
-        self.random_state = random_state
+class OptimizerConfiguration:
+    __conversion_dict = {
+        'tpe': {
+            'gamma': lambda x: float(x),
+            'prior_weight': lambda x: float(x),
+            'n_EI_candidates': lambda x: int(x),
+            "random_state": lambda x: int(x),
+        },
+        "random": {
+            "random_state": lambda x: int(x),
+        }
+    }
 
-    @staticmethod
-    def convertAlgorithmSpec(algorithm_spec):
-        optimizer = OptimizerConfiguration()
+    @classmethod
+    def convert_algorithm_spec(cls, algorithm_spec):
+        ret = {}
+        setting_schema = cls.__conversion_dict[algorithm_spec.algorithm_name]
         for s in algorithm_spec.algorithm_setting:
-            if s.name == "random_state":
-                optimizer.random_state = int(s.value)
-        return algorithm_spec.algorithm_name, optimizer
+            if s.name in setting_schema:
+                ret[s.name] = setting_schema[s.name](s.value)
+
+        return algorithm_spec.algorithm_name, ret
+
+    @classmethod
+    def validate_algorithm_spec(cls, algorithm_spec):
+        algo_name = algorithm_spec.algorithm_name
+        if algo_name == 'tpe':
+            return cls._validate_tpe_setting(algorithm_spec.algorithm_setting)
+        elif algo_name == 'random':
+            return cls._validate_random_setting(algorithm_spec.algorithm_setting)
+        else:
+            return False, "unknown algorithm name %s" % algo_name
+
+    @classmethod
+    def _validate_tpe_setting(cls, algorithm_settings):
+        for s in algorithm_settings:
+            try:
+                if s.name == 'gamma':
+                    if not (1 > float(s.value) > 0):
+                        return False, "gamma should be in the range of (0, 1)"
+                elif s.name == 'prior_weight':
+                    if not (float(s.value) > 0):
+                        return False, "prior_weight should be great than zero"
+                elif s.name == 'n_EI_candidates':
+                    if not (int(s.value) > 0):
+                        return False, "n_EI_candidates should be great than zero"
+                elif s.name == 'random_state':
+                    if not (int(s.value) >= 0):
+                        return False, "random_state should be great or equal than zero"
```

Comment:
[nits] I guess you can remove redundant brackets. It's OK to ignore this comment.

```suggestion
                    if not 1 > float(s.value) > 0:
                        return False, "gamma should be in the range of (0, 1)"
                elif s.name == 'prior_weight':
                    if not float(s.value) > 0:
                        return False, "prior_weight should be great than zero"
                elif s.name == 'n_EI_candidates':
                    if not int(s.value) > 0:
                        return False, "n_EI_candidates should be great than zero"
                elif s.name == 'random_state':
                    if not int(s.value) >= 0:
                        return False, "random_state should be great or equal than zero"
```

Reply:
I only found this commit suggestion can be applied directly after I pushed the same commit = =!

Reply:
No problem 👍 Thanks.